src/ir/possible-contents.cpp

```diff
@@ -1961,25 +2483,41 @@ void Flower::filterExpressionContents(PossibleContents& contents,
                                       const ExpressionLocation& exprLoc,
                                       bool& worthSendingMore) {
   auto type = exprLoc.expr->type;
-  if (!type.isRef()) {
+
+  if (type.isTuple()) {
+    // TODO: Optimize tuples here as well. We would need to take into account
+    //       exprLoc.tupleIndex for that in all the below.
     return;
   }
 
   // The caller cannot know of a situation where it might not be worth sending
   // more to a reference - all that logic is in here. That is, the rest of this
   // function is the only place we can mark |worthSendingMore| as false for a
   // reference.
-  assert(worthSendingMore);
+  bool isRef = type.isRef();
+  if (isRef) {
+    assert(worthSendingMore);
+  }
```

Comment:
```suggestion
  assert(!isRef || worthSendingMore);
```

Can you remind me what "sending more" means in this context?

Reply:
As an optimization, when we know a location cannot increase further, we tell the thing sending it content to stop sending anything more. That avoids wasted work.